tap_github/repository_streams.py

```diff
@@ -1953,3 +1953,114 @@ def _write_schema_message(self):
         if not self._schema_emitted:
             super()._write_schema_message()
             self._schema_emitted = True
+
+
+class DependencyStream(GitHubGraphqlStream):
+    """Defines 'UserContributedToStream' stream. Warning: this stream 'only' gets the first 100 projects (by stars)."""
+
+    name = "dependencies"
+    query_jsonpath = (
+        "$.data.repository.dependencyGraphManifests.nodes.[*].dependencies.nodes.[*]"
+    )
+    primary_keys = ["dependency_repo_id", "repo_id"]
+    parent_stream_type = RepositoryStream
+    state_partitioning_keys = ["repo_id"]
+    ignore_parent_replication_key = True
+
+    @property
+    def http_headers(self) -> dict:
+        """Return the http headers needed.
+
+        Overridden to use the preview for Repository.dependencyGraphManifests.
+        """
+        headers = super().http_headers
+        headers["Accept"] = "application/vnd.github.hawkgirl-preview+json"
+        return headers
+
+    def post_process(self, row: dict, context: Optional[Dict] = None) -> dict:
+        """
+        Add a user_id top-level field to be used as state replication key.
```

Comment:
leftovers from copy/paste?
```suggestion
        Add a dependency_repo_id top-level field to be used as primary key.
```